contracts/src/7683/T1ERC7683.sol

```diff
@@ -1,120 +1,59 @@
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { Address } from "@openzeppelin/contracts/utils/Address.sol";
 import { TypeCasts } from "@hyperlane-xyz/libs/TypeCasts.sol";
-
-import { GaslessCrossChainOrder, ResolvedCrossChainOrder, OnchainCrossChainOrder } from "../interfaces/IERC7683.sol";
-// import { BasicSwap7683 } from "./BasicSwap7683.sol";
-import { Base7683 } from "./Base7683.sol";
 import { Hyperlane7683Message } from "../libraries/7683/Hyperlane7683Message.sol";
 import { OrderData, OrderEncoder } from "../libraries/7683/OrderEncoder.sol";
+
+import { IT1ERC7683 } from "../interfaces/IT1ERC7683.sol";
+import {
+    Output,
+    FillInstruction,
+    GaslessCrossChainOrder,
+    ResolvedCrossChainOrder,
+    OnchainCrossChainOrder
+} from "../interfaces/IERC7683.sol";
+import { T1Permit2 } from "./T1Permit2.sol";
 import { IT1XChainReader } from "../libraries/xChain/IT1XChainReader.sol";
 
-/**
- * @title T1ERC7683
- * @author t1 Labs
- * @notice This contract extends BasicSwap7683 with pull-based settlement using t1 cross-chain reads
- * @dev Implements both push-based messaging and pull-based verification for orders
- */
-contract T1ERC7683 is Base7683, OwnableUpgradeable, PausableUpgradeable {
+/// @title T1ERC7683
+/// @author t1 Labs
+contract T1ERC7683 is IT1ERC7683, T1Permit2, OwnableUpgradeable, PausableUpgradeable {
     using SafeERC20 for IERC20;
 
-    /**
-     * @notice Represents a bid for order settlement
-     * @param settlementReceiver The address that will receive the settlement on src chain
-     * @param amountOut The amount of output tokens to be received on dst chain
-     */
-    struct Bid {
-        address settlementReceiver;
-        uint256 amountOut;
-    }
-
-    // ============ Constants ============
+    /// @notice chain id
     uint32 public immutable localDomain;
     IT1XChainReader public immutable xChainRead;
-    address public counterpart;
 
-    // ============ State Variables ============
+    /// @notice Stores the resolved orders by their ID.
+    mapping(bytes32 orderId => bytes orderData) public openOrders;
+    /// @notice Tracks filled orders and their associated data.
+    mapping(bytes32 orderId => FilledOrder filledOrder) public filledOrders;
+    /// @notice Tracks the status of each order by its ID.
+    mapping(bytes32 orderId => Status status) public orderStatus;
     mapping(bytes32 requestId => uint256 batchIndex) public requestIdToBatchIndex;
-    /// @notice Maps request IDs to order IDs for cross-chain read requests for settlements
-    mapping(bytes32 => bytes32) public settlementReadRequestToOrderId;
     /// @notice Maps request IDs to order IDs for cross-chain read requests for refunds
     mapping(bytes32 => bytes32) public refundReadRequestToOrderId;
-    /// @notice Maps order IDs to verification status
-    mapping(bytes32 => bool) public orderVerified;
```

Comment:
why is this no longer needed?

Reply:
It was not used anywhere in the contract, and redundant with order status from offchain.